### PR TITLE
Add types for [workbox-build]

### PR DESCRIPTION
**Pull request by prichey**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/workbox-build/workbox-build-tests.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    GetManifestConfig,
+    GetManifestResult,
+    getModuleURL,
+} from 'workbox-build';
+
+//==============================================================================
+// WorkboxBuild.copyWorkboxLibraries
+//==============================================================================
+
+export namespace CopyWorkboxLibrariesTest {
```
Comment (jessetrinity) [inline types/workbox-build/workbox-build-tests.ts]: Why does the test export a namespace?

**Review by sandersn — CHANGES_REQUESTED** [review]
Can you try removing the requirement for 3.6 and see if the tests still pass? It would allow a lot more workbox-build consumers to use these types.

(You might also try 3.0 since our telemetry suggests that 99% of people are on 3.0+)

File: types/workbox-build/workbox-build-tests.ts (hunk 1 above)
Comment (prichey) [reply types/workbox-build/workbox-build-tests.ts]: To be honest, I'm not sure. I was copying the style in the other [workbox] types, e.g. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/workbox-core/workbox-core-tests.ts. I can remove if this is an issue

File: types/workbox-build/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for workbox-build 5.0
+// Project: https://github.com/GoogleChrome/workbox/tree/master/packages/workbox-build, https://developers.google.com/web/tools/workbox/modules/workbox-build
+// Definitions by: Preston Richey <https://github.com/prichey>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.6
```
[the remark this answers is not in the document]
Comment (prichey) [reply types/workbox-build/index.d.ts]: I'll give this a shot.

**Review by sandersn — APPROVED** [review]

**Comment by igorovic** [comment]
I am using Workbox 6.1.0 and the type definition for workbox-build is broken since workbox-core provides the type definition for 
`RouteHandlerCallback` and  `RouteMatchCallback`


I had to manually remove `@types/workbox-routing` and modifdy the file `@types/workbox-build/_types.d.ts` with the following. 

**before**
```ts
import { RouteHandlerCallback, RouteMatchCallback } from 'workbox-routing';
```

**after**

```ts
import { RouteHandlerCallback, RouteMatchCallback } from 'workbox-core';
```

`@types/workbox-build` doesn't depend on `@types/workbox-routing`

**Pull request merged**